sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -35,6 +40,9 @@
   /** Class of the original {@link DoFn} from which this signature was produced. */
   public abstract Class<? extends DoFn<?, ?>> fnClass();
 
+  /** Whether this {@link DoFn} does a bounded amount of work per element. */
+  public abstract PCollection.IsBounded isBounded();
```

Comment:
I think `boundedWorkPerElement` would be clearer, despite the verbosity. I don't know what it means, otherwise, for a `DoFn` to be bounded (I do know what it means for a function to be bounded, but that is not the case here).